lib/segment/src/index/query_estimator.rs

```diff
@@ -125,6 +125,27 @@ pub fn combine_should_estimations(
     }
 }
 
+pub fn combine_min_should_estimations(
+    estimations: &[CardinalityEstimation],
+    min_count: usize,
+    total: usize,
+) -> CardinalityEstimation {
+    /*
+    | First estimate cardinality of intersections and then combine the estimations
+    | ex) min_count : 2, # of estimations : 4
+    | |(A ⋂ B) ∪ (A ⋂ C) ∪ (A ⋂ D) ∪ (B ⋂ C) ∪ (B ⋂ D) ∪ (C ⋂ D)|
+     */
+    let intersection_estimations = estimations
+        .iter()
+        .combinations(min_count)
+        .map(|intersection| {
+            combine_must_estimations(&intersection.into_iter().cloned().collect_vec(), total)
+        })
+        .collect_vec();
+
+    combine_should_estimations(&intersection_estimations, total)
+}
+
```

Comment:
May I ask for suggestions or feedback about cardinality estimation?

Reply:
After thinking about it for a while, I believe the logic is sound. I also added an extra equivalence test for completeness.